File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/model/ResultVo.java
package org.javamaster.invocationlab.admin.model;

import lombok.AllArgsConstructor;
import lombok.Data;

/**
 * @author yudong
 * @date 2023/2/14
 */
@Data
@AllArgsConstructor
public class ResultVo<T> {
    private Integer code;
    private String msg;
    private T data;

    public static <T> ResultVo<T> success(T data) {
        return new ResultVo<>(200, "操作成功", data);
    }

    public static <T> ResultVo<T> fail(String msg) {
        return new ResultVo<>(400, msg, null);
    }

    public static <T> ResultVo<T> fail(Integer code, String msg) {
        return new ResultVo<>(code, msg, null);
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/model/ResultVo.kt
package org.javamaster.invocationlab.admin.model

import org.javamaster.invocationlab.admin.annos.AllOpen

/**
 * @author yudong
 * @date 2023/2/14
 */

@AllOpen
class ResultVo<T>(var code: Int? = null, var msg: String? = null, val data: T? = null) {

    companion object {
        fun <T> success(data: T): ResultVo<T> {
            return ResultVo(200, "操作成功", data)
        }

        fun <T> fail(msg: String?): ResultVo<T> {
            return ResultVo(400, msg, null)
        }

        fun <T> fail(code: Int?, msg: String?): ResultVo<T> {
            return ResultVo(code, msg, null)
        }
    }

}
